emu_base/math/krylov_energy_min.py

```diff
@@ -8,23 +8,33 @@ def _eigen_pair(
     T_trunc: torch.Tensor,
     guessed_state: torch.Tensor,
     residual_tolerance: float,
+    is_hermitian: bool,
 ) -> Tuple[torch.Tensor, torch.Tensor]:
     """
-    Return the smallest (k = 1) eigenpair of T_trunc.
-    If T_trunc is too small for LOBPCG (n < 3), do a full eigh;
-    otherwise call lobpcg with the given initial guessed state.
+    Return the smallest eigenpair of T_trunc.
+    If T_trunc is small (n < 3) or Hermitian, do a full eigh;
+    otherwise use lobpcg with the given initial guessed state.
```

Comment:
change a bit the description. The code uses `lobpcg` for large hermitean matrices.

It seems that it will be good to join what `eigh` and `eig` do here. `eigh` vs `eig` performance gain for `n==2` is not significant